### PR TITLE
pythonPackages: Add pyocclient and dash bootstrap components

### PR DESCRIPTION
**Pull request by turion**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by turion** [comment]
@ofborg build python3Packages.pyocclient python3Packages.dash-bootstrap-components

**Comment by turion** [comment]
Will merge in the next days if there are no objections.

**Comment by neosimsim** [comment]
lgtm

**Review by neosimsim — APPROVED** [review]

File: pkgs/development/python-modules/pyocclient/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyocclient/default.nix:28]: Source should be between version and the used inputs.

File: pkgs/development/python-modules/pyocclient/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyocclient/default.nix:17]: This needs to be propagatedBuildInputs.

File: pkgs/development/python-modules/dash-bootstrap-components/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dash-bootstrap-components/default.nix:26]: Source should be between version and the used inputs.

File: pkgs/development/python-modules/dash-bootstrap-components/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dash-bootstrap-components/default.nix:13]: This needs to be propagatedBuildInputs.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/dash-bootstrap-components/default.nix
Comment (Janik-Haag) [reply pkgs/development/python-modules/dash-bootstrap-components/default.nix:26]: resolving this since it was posted twice.

File: pkgs/development/python-modules/dash-bootstrap-components/default.nix
Comment (Janik-Haag) [reply pkgs/development/python-modules/dash-bootstrap-components/default.nix:13]: resolving this since it was posted twice.

**Comment by Janik-Haag** [comment]
@turion can you please apply the changes sandro suggested and rebase so we can get this merged?

**Comment by SuperSandro2000** [comment]
This PR has been stale for a year now, closing

**Comment by turion** [comment]
Rebased and addressed all concerns.

File: pkgs/development/python-modules/pyocclient/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyocclient/default.nix:36]:
```suggestion
    changelog = "https://github.com/owncloud/pyocclient/blob/v${version}/docs/source/CHANGES.rst";
```
that renders in the browser :)

File: pkgs/development/python-modules/dash-bootstrap-components/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dash-bootstrap-components/default.nix:35]:
```suggestion
    changelog = "https://github.com/facultyai/dash-bootstrap-components/blob/r${version}/docs/content/changelog.md";
```

File: pkgs/development/python-modules/dash-bootstrap-components/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dash-bootstrap-components/default.nix:22]: Did you try with pytestCheckHook?

**Review by SuperSandro2000 — COMMENTED** [review]
Can you fix the commits to include python310Packages so that ofborg works for them and they don't look like they are for python2?

File: pkgs/development/python-modules/dash-bootstrap-components/default.nix
Comment (SuperSandro2000) [reply pkgs/development/python-modules/dash-bootstrap-components/default.nix:22]: Try getting the source from github

**Comment by kirillrdy** [comment]
Result of `nixpkgs-review pr 177038` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.dash-bootstrap-components</li>
    <li>python310Packages.dash-bootstrap-components.dist</li>
    <li>python310Packages.pyocclient</li>
    <li>python310Packages.pyocclient.dist</li>
    <li>python311Packages.dash-bootstrap-components</li>
    <li>python311Packages.dash-bootstrap-components.dist</li>
    <li>python311Packages.pyocclient</li>
    <li>python311Packages.pyocclient.dist</li>
  </ul>
</details>

@SuperSandro2000 what is outstanding ? Can I help ?

File: pkgs/development/python-modules/dash-bootstrap-components/default.nix
Comment (SuperSandro2000) [reply pkgs/development/python-modules/dash-bootstrap-components/default.nix:22]: I think this is one of the last left open things

**Comment by turion** [comment]
@kirillrdy if you want you can make yourself maintainer and finish the PR :) the last open thing is: https://github.com/NixOS/nixpkgs/pull/177038#discussion_r1271557046

**Comment by turion** [comment]
I've since lost interest in maintaining these packages.

**Pull request closed without merging**
